### PR TITLE
Add posh_ps_set_acl

### PR DESCRIPTION
**Pull request by frack113**
### Summary of the Pull Request

Detect when a powershell script set acl in windows folder or subfolder

### Detailed Description of the Pull Request / Additional Comments

Atomic test failled but no one should change ACL in windows in a daily live

### Example Log Event

`Set-Acl -Path "C:\Windows\System32\termsrv.dll" -AclObject $ACL`

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

File: rules/windows/powershell/powershell_script/posh_ps_set_acl.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+title: Powershell Set-Acl Windows Folder
+id: cae80281-ef23-44c5-873b-fd48d2666f49
+status: experimental
+description: Detects powershell scripts set ACL to a file in the Windows folder
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl.yml:4]:
```suggestion
description: Detects PowerShell scripts that set ACLs to a file in the Windows folder
```

File: rules/windows/powershell/powershell_script/posh_ps_set_acl.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+title: Powershell Set-Acl Windows Folder
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl.yml:1]:
```suggestion
title: PowerShell Set-Acl Windows Folder
```

File: rules/windows/powershell/powershell_script/posh_ps_set_acl.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+logsource:
+    product: windows
+    category: ps_script
+    definition: bade5735-5ab0-4aa7-a642-a11be0e40872
+detection:
+    selection:
+        ScriptBlockText|contains|all:
+            - 'Set-Acl '
+            - '-Path '
+            - 'Windows'
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl.yml:21]:
```suggestion
            - :'\Windows\'
```
just to be sure? I don't see a big downside and a reduce in possible FPs

**Review by phantinuss — APPROVED** [review]

File: rules/windows/powershell/powershell_script/posh_ps_set_acl.yml (hunk 3 above)
Comment (frack113) [reply rules/windows/powershell/powershell_script/posh_ps_set_acl.yml:21]: image
Can play with `\` or `/` in powershell on windows 11.
so more a 
```yaml
selection_path:
    - ':\Windows'
    - ':/Windows'
condition all of selection_*
```
What do you think ?

Comment (frack113) [reply rules/windows/powershell/powershell_script/posh_ps_set_acl.yml]: Done

**Comment by nasbench** [comment]
I put a WIP :D so no one changes because I was changing :)

File: rules/windows/powershell/powershell_script/posh_ps_set_acl.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            - 'Set-Acl '
+            - '-Path '
+    selection_path:
+        ScriptBlockText|contains:
+            - ':\Windows\'
+            - ':\Windows/'
+            - ':/Windows\'
+            - ':/Windows/'
+            - '$env:windir\'
+            - '$env:windir/'
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl.yml:28]:
```suggestion
            - '$env:windir'
            - '$env:windir'
```
I think here we don't necessarily need it. (there is no `$env:windirsomethingelse`)

**Review by phantinuss — APPROVED** [review]

**Comment by nasbench** [comment]
Just added process creation variations. And made a generic split for basic ACL changes and added constraints to the high one.

**Review by nasbench — APPROVED** [review]

File: rules/windows/powershell/powershell_script/posh_ps_set_acl_susp_location.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            - 'Set-Acl '
+            - '-AclObject '
+    selection_paths:
+        # Note: Add more suspicious paths
+        ScriptBlockText|contains:
+            - '-Path "C:\Windows'
+            - '-Path "C:/Windows'
+            - "-Path 'C:\\Windows"
+            - "-Path 'C:/Windows"
+            - '-Path $env:windir'
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl_susp_location.yml:36]:
```suggestion
            - '-Path "C:\Windows'
            - '-Path "C:/Windows'
            - "-Path 'C:\\Windows"
            - "-Path 'C:/Windows"
            - '-Path C:\\Windows'
            - '-Path C:/Windows'
            - '-Path $env:windir'
            - '-Path "$env:windir'
            - "-Path '$env:windir"
```
$windir subpaths rarely have a space, so quoting is optional, isn't it?

File: rules/windows/process_creation/proc_creation_win_powershell_set_acl.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+id: bdeb2cff-af74-4094-8426-724dc937f20a
+related:
+    - id: cae80281-ef23-44c5-873b-fd48d2666f49 # PsScript Low
+      type: derived
+    - id: 0944e002-e3f6-4eb5-bf69-3a3067b53d73 # ProcCreation Susp
+      type: derived
+    - id: 3bf1d859-3a7e-44cb-8809-a99e066d3478 # PsScript High
+      type: derived
+status: experimental
+description: Detects powershell execution to set the ACL of a file or a folder
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_set_acl.yml:11]:
```suggestion
description: Detects PowerShell execution to set the ACL of a file or a folder
```

File: rules/windows/process_creation/proc_creation_win_powershell_set_acl.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
+title: Powershell Script Change Permission Via Set-Acl
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_set_acl.yml:1]:
```suggestion
title: PowerShell Script Change Permission Via Set-Acl
```

File: rules/windows/powershell/powershell_script/posh_ps_set_acl_susp_location.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+title: Powershell Set-Acl On Windows Folder - PsScript
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl_susp_location.yml:1]:
```suggestion
title: PowerShell Set-Acl On Windows Folder - PsScript
```

File: rules/windows/powershell/powershell_script/posh_ps_set_acl_susp_location.yml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+id: 3bf1d859-3a7e-44cb-8809-a99e066d3478
+related:
+    - id: cae80281-ef23-44c5-873b-fd48d2666f49 # PsScript Low
+      type: derived
+    - id: 0944e002-e3f6-4eb5-bf69-3a3067b53d73 # ProcCreation Susp
+      type: derived
+    - id: bdeb2cff-af74-4094-8426-724dc937f20a # ProcCreation Low
+      type: derived
+status: experimental
+description: Detects powershell scripts to set the ACL to a file in the Windows folder
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl_susp_location.yml:11]:
```suggestion
description: Detects PowerShell scripts to set the ACL to a file in the Windows folder
```

File: rules/windows/powershell/powershell_script/posh_ps_set_acl.yml [hunk 10]
```diff
@@ -0,0 +1,32 @@
+title: Powershell Script Change Permission Via Set-Acl - PsScript
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl.yml:1]:
```suggestion
title: PowerShell Script Change Permission Via Set-Acl - PsScript
```

File: rules/windows/powershell/powershell_script/posh_ps_set_acl.yml [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+id: cae80281-ef23-44c5-873b-fd48d2666f49
+related:
+    - id: 0944e002-e3f6-4eb5-bf69-3a3067b53d73 # ProcCreation Susp
+      type: derived
+    - id: bdeb2cff-af74-4094-8426-724dc937f20a # ProcCreation Low
+      type: derived
+    - id: 3bf1d859-3a7e-44cb-8809-a99e066d3478 # PsScript High
+      type: derived
+status: experimental
+description: Detects powershell scripts set ACL to of a file or a folder
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_script/posh_ps_set_acl.yml:11]:
```suggestion
description: Detects PowerShell scripts set ACL to of a file or a folder
```

File: rules/windows/process_creation/proc_creation_win_powershell_set_acl_susp_location.yml [hunk 12]
```diff
@@ -0,0 +1,48 @@
+title: Powershell Set-Acl On Windows Folder
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_set_acl_susp_location.yml:1]:
```suggestion
title: PowerShell Set-Acl On Windows Folder
```

File: rules/windows/process_creation/proc_creation_win_powershell_set_acl_susp_location.yml [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+id: 0944e002-e3f6-4eb5-bf69-3a3067b53d73 # ProcCreation Susp
+related:
+    - id: cae80281-ef23-44c5-873b-fd48d2666f49 # PsScript Low
+      type: derived
+    - id: bdeb2cff-af74-4094-8426-724dc937f20a # ProcCreation Low
+      type: derived
+    - id: 3bf1d859-3a7e-44cb-8809-a99e066d3478 # PsScript High
+      type: derived
+status: experimental
+description: Detects powershell scripts to set the ACL to a file in the Windows folder
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_set_acl_susp_location.yml:11]:
```suggestion
description: Detects PowerShell scripts to set the ACL to a file in the Windows folder
```

File: rules/windows/process_creation/proc_creation_win_powershell_set_acl_susp_location.yml [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    selection_cmdlet:
+        CommandLine|contains|all:
+            - 'Set-Acl '
+            - '-AclObject '
+    selection_paths:
+        # Note: Add more suspicious paths
+        CommandLine|contains:
+            - '-Path "C:\Windows'
+            - "-Path 'C:\\Windows"
+            - '-Path %windir%'
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_set_acl_susp_location.yml:39]:
```suggestion
            - '-Path "C:\Windows'
            - "-Path 'C:\\Windows"
            - '-Path %windir%'
            - '-Path $env:windir'
```
The parent who calls powershell with `/command` can be anything. Also another powershell. So we would need all the permutations again plus the batch stuff.

**Review by phantinuss — APPROVED** [review]

File: rules/windows/powershell/powershell_script/posh_ps_set_acl_susp_location.yml (hunk 5 above)
Comment (nasbench) [reply rules/windows/powershell/powershell_script/posh_ps_set_acl_susp_location.yml:36]: True

**Pull request merged**